cmd/kubelet/app/server.go

```diff
@@ -839,7 +839,7 @@ func run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Depend
 				TopologyManagerScope:                    s.TopologyManagerScope,
 				TopologyManagerPolicyOptions:            topologyManagerPolicyOptions,
 			},
-			s.FailSwapOn,
+			s.KubeletConfiguration.FailSwapOn,
```

Comment:
the test is correct, the flag is true

> Jan 10 16:09:36 kind-control-plane kubelet[324]: I0110 16:09:36.749688     324 flags.go:64] FLAG: --fail-swap-on="true"

the problem is that here you are setting the value of the configuration

This is a breaking change, you can not change the order of preference of flags vs configuration so lightly

Reply:
Well I logged what s.FailSwapOn is and its false in this case. So this is probably not being set..

Reply:
We are parsing this bool from the cli and for some reason its set to false (or not being set at all).